src/t_zset.c

```diff
@@ -115,15 +115,40 @@ static inline void zslSetNodeHeight(zskiplistNode *x, int height) {
 }
 
 /* Create a skiplist node with the specified number of levels.
- * The SDS string 'ele' is referenced by the node after the call. */
+ * By embedding elements and levels into the skiplist nodes,
+ * we achieve better cache-friendliness and a more compact memory structure.
+ *
+ * The memory layout is as follows:
+ *
+ *     +-------+------------------+---------+-----+---------+-------------+
+ *     | score | backward-pointer | level-0 | ... | level-N | element-sds |
+ *     +-------+------------------+---------+-----+---------+-------------+
+ */
 static zskiplistNode *zslCreateNode(int height, double score, sds ele) {
```

Comment:
For functions that copy the sds, it's possible to use the `const_sds` type.

```suggestion
static zskiplistNode *zslCreateNode(int height, double score, const_sds ele) {
```